Drop Python 3.5

sirosen opened the pull request:
Also update pre-commit config and relevant docs.

There are also a bunch of changes from pyupgrade.

closes #543

sirosen committed:
Drop Python 3.5

Also update pre-commit config and relevant docs.

lafrech commented on src/webargs/asyncparser.py:
```diff
@@ -1,4 +1,4 @@
-"""Asynchronous request parser. Compatible with Python>=3.5."""
+"""Asynchronous request parser. Compatible with Python>=3.6."""
```
This Python version compat comment should be removed.

lafrech reviewed (commented):
Thanks for this !

TODO

- Remove comment in async parser
- Update tox.ini: remove py35 (and perhaps update marshmallow>=3.0.0rc2, unrelated but while we're at it)

sirosen committed:
Update tox.ini to remove 3.5, update ma version

Remove 3.5 from tox.ini because it's no longer supported.
Also, while we're here, switch '3.0.0rc2' specifier to '3.0.0'

Remove old comment about asyncparser compatibility.

sirosen commented:
I knew I must have missed something! 😄 
I'm guessing that comment was added for python2 users, but I wasn't sure.

I've pushed a commit with the changes.

lafrech commented:
Good. Thanks.

Yes, that comment dates back to when Python2 was still supported and the async parser was in a dedicated py3 directory.

Approved. :+1:

sloria commented on .pre-commit-config.yaml:
```diff
@@ -1,15 +1,15 @@
   hooks:
   - id: pyupgrade
-    args: ["--py3-plus", "--keep-mock"]
+    args: ["--py36-plus", "--keep-mock"]
```
Do we still need `--keep-mock`?

sirosen replied:
Good catch! I don't think it does anything now that we're not using the backport. I waited to reply until I got to confirm this.
I just opened #556 to clean up.
